### PR TITLE
GradientPicker: remove padding and disable overflow of color picker popovers

### PR DESCRIPTION
**Pull request by ciampo**
## What?

As flagged by @t-hamano in https://github.com/WordPress/gutenberg/pull/55149#issuecomment-1757673201, popovers containing the `ColorPicker` in the `GradientPicker` component include some padding around the color picker. This PR removes the padding (and overflow: hidden styles) to have the popover look more uniform with other similar color picker popovers.

## Why?

UI polish and consistency

## How?

- Remove some `ColorPicker`-specific styles
- Introduce the `resize: false` popover prop (which also causes `overflow: hidden` styles to be applied)
- Use the `DropdownContentWrapper` component to easily remove the popover's internal padding

## Testing Instructions

- Open the `GradientPicker` Storybook example
- Click on the color stops on the gradient bar
- Notice how the popover containing the gradient picker doesn't have internal padding, and how the drag handle correctly overflows its parent
- 
## Screenshots or screencast 

| Before (`trunk`) | After (this PR) |
|---|---|
|  Screenshot 2023-10-11 at 16 52 07 | Screenshot 2023-10-11 at 16 53 26

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 -   `Notice`: Remove margins from `Notice` component ([#54800](https://github.com/WordPress/gutenberg/pull/54800)).
 -   `ToolsPanel`: do not apply the `className` to prop to `ToolsPanelItem` components when rendered as placeholders ([#55207](https://github.com/WordPress/gutenberg/pull/55207)).
+- `GradientPicker`: remove overflow styles and padding from `ColorPicker` popovers ([#55265](https://github.com/WordPress/gutenberg/pull/55265)).
```
Comment (t-hamano) [inline packages/components/CHANGELOG.md]:
```suggestion
-   `GradientPicker`: remove overflow styles and padding from `ColorPicker` popovers ([#55265](https://github.com/WordPress/gutenberg/pull/55265)).
```

Nit

File: packages/components/src/color-picker/styles.ts [hunk 2]
```diff
@@ -81,7 +81,6 @@ export const ColorfulWrapper = styled.div`
 		align-items: center;
 		width: 216px;
 		height: auto;
-		overflow: hidden;
```
Comment (t-hamano) [inline packages/components/src/color-picker/styles.ts]: If this style is removed, an unintended scroll bar will appear when I move the pointer to the right edge of the regular color picker. This problem can be reproduced on Windows, but it might be possible to reproduce it on MacOS by changing the scroll bar settings.

However, strangely, this problem does not occur in #55149, where [the same style has been removed](https://github.com/WordPress/gutenberg/pull/55149/files#diff-28b8d7edaea447b5f40b2b02ba5fa261ce5164360c75a067e47612276f4bd568L84).

https://github.com/WordPress/gutenberg/assets/54422211/e859843e-8fa6-4030-aea3-2495eed8a1df

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR!

It's working as expected, but I'd like to leave a few comments.

File: packages/components/src/color-picker/styles.ts (hunk 2 above)
Comment (ciampo) [reply packages/components/src/color-picker/styles.ts]: I think that's because it's missing the change to `packages/components/src/color-palette/index.tsx` setting `resize: false` for that specific instance of dropdown.

This can be confirmed by removing the `resize : false` change in this PR in the `packages/components/src/custom-gradient-picker/gradient-bar/control-points.tsx` file, and noting how the popover in `GradientPicker` will also start scrolling.

In short, I think this is behaving as expected.

**Review by brookewp — APPROVED** [review]
Looks great! 🚀
